addons/mail/static/src/core/messaging_service.js

```diff
@@ -635,6 +643,18 @@ export class Messaging {
         }
     }
 
+    _removeThread(threadData) {
+        const thread = this.threadService.insert({
+            ...threadData,
+            model: "discuss.channel",
+        });
+        this.threadService.remove(thread);
+        if (thread.localId === this.store.discuss.threadLocalId) {
+            this.store.discuss.threadLocalId = undefined;
+        }
+        return thread;
+    }
```

Comment:
This function is not worth it. It's very small, and introducing it makes it confusing (e.g. difference between `threadService.remove()` and `_removeThread()`).